src/tests/cmocka/test_ad_gpo.c

```diff
@@ -401,6 +401,19 @@ void test_ad_gpo_parse_sd(void **state)
     talloc_free(sd);
 }
 
+errno_t ad_gpo_parse_ini_file(const char *smb_path, int *_gpt_version);
+
+void test_ad_gpo_parse_ini_file(void **state)
+{
+    int version = -1;
+
+    ad_gpo_parse_ini_file(
+        REVERSE_GPO_CACHE_PATH  ABS_SRC_DIR "/src/tests/cmocka/",
+        &version);
+
+    assert_int_equal(version, 6);
```

Comment:
I'd also expect that you check the value of `displayName` to be a sequence of `?` characters. This way you assure that the algorithm implemented in `gpo_sanitize_buffer_content()` is working correctly.

Reply:
The purpose of the test is to make sure `ad_gpo_parse_ini_file()` can process files in non-UTF encoding.

Using of '?' is implementation detail. It might happen that one day we decide to change implementation to convert content to utf-8 instead.

Reply:
I see. Thanks for the clarification.